packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1556,19 +1703,27 @@ describe("BnsConnection", () => {
     // ----- connection.searchTx() -----
     // we should be able to find the transaction through quite a number of tag queries
 
-    const txById = await connection.searchTx({ tags: [bnsSwapQueryTags(querySwapId)] });
+    const txById = (await connection.searchTx({ tags: [bnsSwapQueryTags(querySwapId)] })).filter(
+      isConfirmedTransaction,
+    );
     expect(txById.length).toEqual(1);
     expect(txById[0].transactionId).toEqual(transactionId);
 
-    const txBySender = await connection.searchTx({ tags: [bnsSwapQueryTags(querySwapSender)] });
+    const txBySender = (await connection.searchTx({ tags: [bnsSwapQueryTags(querySwapSender)] })).filter(
+      isConfirmedTransaction,
+    );
     expect(txBySender.length).toBeGreaterThanOrEqual(1);
     expect(txBySender[txBySender.length - 1].transactionId).toEqual(transactionId);
 
-    const txByRecipient = await connection.searchTx({ tags: [bnsSwapQueryTags(querySwapRecipient)] });
+    const txByRecipient = (await connection.searchTx({
+      tags: [bnsSwapQueryTags(querySwapRecipient)],
+    })).filter(isConfirmedTransaction);
     expect(txByRecipient.length).toEqual(1);
     expect(txByRecipient[0].transactionId).toEqual(transactionId);
 
-    const txByHash = await connection.searchTx({ tags: [bnsSwapQueryTags(querySwapHash)] });
+    const txByHash = (await connection.searchTx({ tags: [bnsSwapQueryTags(querySwapHash)] })).filter(
```

Comment:
If swapQueryTags are related to state defined in `bcp/*` transactions not just `bns/*` specific ones, we should probably define a generic way to create them in the `searchTx` query parameters. Just like `sentToOrFrom`.

This can be a different PR, but I figure a minor update for 0.11 (or 0.12)